docs/source/gateway/index.rst

```diff
@@ -1,6 +1,1049 @@
+The MLflow AI Gateway helps mitigate these issues by centrally managing the keys and allowing users to
+access the service without ever needing to handle API keys directly.
+
+Here's a brief overview of how the authentication process works:
+
+* **Key Management**: As an administrator, you'll set up the MLflow AI Gateway and input your API keys (as environment variables or directly into the configuration files). The Gateway securely stores these keys and uses them to authenticate requests with the service provider.
+
+* **User Access**: Instead of providing individual API keys to every user in your organization, you'll give them access to the MLflow AI Gateway. Users then send their requests directly to the Gateway, which acts as a proxy to the service provider.
+
+* **Request Handling**: When a user sends a request to the MLflow AI Gateway, it takes the request, adds the necessary authentication (using the API keys it manages), and forwards the request to the correct provider (as specified in the configuration files).
+
+* **Response Forwarding**: The Gateway receives the response from the provider and then sends this response back to the user.
+
+By centralizing key management, the MLflow AI Gateway dramatically reduces the risk of keys being
+lost, misused, or accessed by unauthorized individuals. This approach also simplifies the process
+for end users - they no longer need to worry about managing API keys and can instead focus on
+making requests and working with the responses.
+
+This centralized system allows you to change providers or models easily. If you need
+to switch models or update API keys, you can do so in a central location without requiring any
+changes from your end users. This makes the MLflow AI Gateway a robust and versatile solution for
+API key management and service integration within your organization.
+
+AI Gateway Server
+-----------------
+For some users, it might be preferable to interface directly with the MLflow AI Gateway server using the REST API.
+This provides flexibility and allows for a broader range of interactions that may not be covered by the ``client`` or ``fluent`` APIs.
+
+FastAPI Documentation ("/docs")
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+FastAPI, the framework used for building the MLflow AI Gateway, provides an automatic interactive API
+documentation interface, which is accessible at the "/docs" endpoint (e.g., "http://my.gateway:9000/docs").
+This interactive interface is very handy for exploring and testing the available API endpoints.
+
+As a convenience, accessing the root URL (e.g., "http://my.gateway:9000") redirects to this "/docs" endpoint.
+
+Gateway Health ("/health")
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+The "/health" endpoint (e.g., "http://my.gateway:9000/health") is used for health checking the
+Gateway Server. It returns a 200 OK HTTP response if the server is running and healthy.
+This endpoint is particularly useful for monitoring and alerting systems that check service health.
```

Comment:
I think we can remove this. Most users probably won't call it, and folks who really care will find it in the `/docs`